src/app/checkout/step-3/step-3.component.js

```diff
@@ -90,14 +92,17 @@ class Step3Controller{
       submitRequest = Observable.throw('Current payment type is unknown');
     }
     submitRequest.subscribe(() => {
+        this.submittingOrder = false;
         this.orderService.clearCardSecurityCode();
         this.onSubmitted();
         this.$window.location.href = 'thank-you.html';
       },
       (error) => {
+        this.submittingOrder = false;
         this.$log.error('Error submitting purchase:', error);
         this.onSubmitted();
         // TODO: show error message
+        this.errors = [error];
```

Comment:
Does this resolve the TODO comment from the line above, or does more need to be done to show errors?